fetch.js

```diff
@@ -216,6 +233,10 @@
       }
 
       this.text = function() {
+        if (this.body) {
+          return this.blob().then(readBlobAsText)
```

Comment:
We have some checks in place (see `consumed()` below) to prevent reading the body data multiple times. Should the same restrictions also apply to readable streams?